### ISSUE TITLE
Arm64: "no image found" error on arm64

### ISSUE THREAD

**Issue by Pennyzct** [containers/image#898]
Hi guys

When I launch a k8s v1.17.x/crio v1.17.3 cluster on ARM node, the following errors occurred:
```
[ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-controller-manager:v1.17.3: output: time="2020-04-16T15:09:47+08:00" level=fatal msg="pulling image failed: rpc error: code = Unknown desc = Error choosing image instance: no image found in manifest list for architecture arm64, variant v8, OS linux"
, error: exit status 1
```
IMO, in https://github.com/containers/image/pull/854, we imported absolute variant check when pulling multi-arch images and it brought errors when arm image entry doesn't have variant info. Like `k8s.gcr.io/kube-controller-manager:v1.17.3`
```
$ ./manifest-tool-linux-arm64 inspect k8s.gcr.io/kube-controller-manager:v1.17.3
Name:   k8s.gcr.io/kube-controller-manager:v1.17.3 (Type: application/vnd.docker.distribution.manifest.list.v2+json)
Digest: sha256:2f0bf4d08e72a1fd6327c8eca3a72ad21af3a608283423bb3c10c98e68759844
 * Contains 5 manifest references:
1    Mfst Type: application/vnd.docker.distribution.manifest.v2+json
1       Digest: sha256:8f03391c0d22e3da8d22725178efe3c4338e1920504f7f4eb4a1c7f5f40c4c6e
1  Mfst Length: 741
1     Platform:
1           -      OS: linux
1           - OS Vers:
1           - OS Feat: []
1           -    Arch: amd64
1           - Variant:
1           - Feature:
1     # Layers: 2
         layer 1: digest = sha256:597de8ba0c30cdd0b372023aa2ea3ca9b3affbcba5ac8db922f57d6cb67db7c8
         layer 2: digest = sha256:02c23a6c0b486ac6845f53754f33dbf44c0807ac45fd7ea83f7f4c7943674888

2    Mfst Type: application/vnd.docker.distribution.manifest.v2+json
2       Digest: sha256:fd2317542b55c3603c79644bd6eb86b39721f9d20989f6fe31a5e15533e400fb
2  Mfst Length: 741
2     Platform:
2           -      OS: linux
2           - OS Vers:
2           - OS Feat: []
2           -    Arch: arm
2           - Variant:
2           - Feature:
2     # Layers: 2
         layer 1: digest = sha256:57c903ae07d51423a146c81279de66a7778633e3021682eb6efe556ad2e01cd6
         layer 2: digest = sha256:b472c73998329f32b25a341a914330829cc707bc7b5683764df3d49ad42d06fe

3    Mfst Type: application/vnd.docker.distribution.manifest.v2+json
3       Digest: sha256:415ffee927d375798ffa3d1930528bbef3f745e6f01063a69c12a95992d7ed73
3  Mfst Length: 741
3     Platform:
3           -      OS: linux
3           - OS Vers:
3           - OS Feat: []
3           -    Arch: arm64
3           - Variant:
3           - Feature:
3     # Layers: 2
         layer 1: digest = sha256:ed2e7fd674160401fd6455ab65b27f66648ac61ae9698c73cd70c19943b900ce
         layer 2: digest = sha256:4cd08458fe9f2baadbd994c2e7d77e0d2001b435b19af50ba641d4f59b9d42fe

4    Mfst Type: application/vnd.docker.distribution.manifest.v2+json
4       Digest: sha256:0a10bf1db7491900b5900eb9bd7021769774565b483cd86512a479f66478267e
4  Mfst Length: 741
4     Platform:
4           -      OS: linux
4           - OS Vers:
4           - OS Feat: []
4           -    Arch: ppc64le
4           - Variant:
4           - Feature:
4     # Layers: 2
         layer 1: digest = sha256:6790af3178f26fef1de09371d9ecd402d1bbb42a2fdf1b9421bcc590be812a1e
         layer 2: digest = sha256:5357ae8bd88b145b5510d6544486cbc2f47029b3d179c9f34b0fa281b2313e55

5    Mfst Type: application/vnd.docker.distribution.manifest.v2+json
5       Digest: sha256:2a38bd5eaff2cb02163a94b44cd6926d59b1085b9dc3008098adfb3fd9470344
5  Mfst Length: 741
5     Platform:
5           -      OS: linux
5           - OS Vers:
5           - OS Feat: []
5           -    Arch: s390x
5           - Variant:
5           - Feature:
5     # Layers: 2
         layer 1: digest = sha256:29aaea6da3fd5d46bae11d28f2f3f69bc5da18e6b74b8256dcc71d9ba7684d35
         layer 2: digest = sha256:f58e055681530917d477173d2699330fbb44a386a5938998fad5d3a0d886836b
```

**Suggestion:**

How about relaxing the variant check on arm64, since that for now, we only have v8 for `arm64/aarch64` ?
 or importing **platform normalization** scheme here to normalize and complement platform info, see https://github.com/containerd/containerd/blob/master/platforms/platforms.go#L269

**Comment by saschagrunert** [comment]
I know that arm v8 is compatible with v7 when running on Linux, but does this also apply to the other variants? In other words: Can we agree that we do not provide a v8 variant for arm if we have a v7 one? If yes, then we should relax the matching a bit.

**Comment by Pennyzct** [comment]
Hi @saschagrunert 

yep, the variant should be backward compatible. ;)  
> Can we agree that we do not provide a v8 variant for arm if we have a v7 one?
 
yes, IMO, we already provided backward compatibility here, see https://github.com/containers/image/blob/master/internal/pkg/platform/platform_matcher.go#L149

What I mentioned here, is that, if the arm64 entry in multi-arch image doesn't provide variant info, and we detected our host info with `arm64v8`, which leading us failing to pull the image.
IMO, Since `v8` is the only major variant option for arm 64bit machine, for now.  maybe we could relax/complement the variant info to `v8` ? ;)
